src/hyperfine/format.rs

```diff
@@ -16,11 +44,54 @@ pub fn format_duration(duration: Second, unit: Option<Unit>) -> String {
 
 /// Like `format_duration`, but returns the target unit as well.
 pub fn format_duration_unit(duration: Second, unit: Option<Unit>) -> (String, Unit) {
-    if (duration < 1.0 && unit.is_none()) || unit == Some(Unit::MilliSecond) {
-        (format!("{:.1} ms", duration * 1e3), Unit::MilliSecond)
-    } else {
-        (format!("{:.3} s", duration), Unit::Second)
+    let (out_str, out_unit) = format_duration_value(duration, unit);
+
+    (format!("{} {}", out_str, out_unit.short_name()), out_unit)
+}
+
+/// Like `format_duration`, but returns the target unit as well.
+pub fn format_duration_value(duration: Second, unit: Option<Unit>) -> (String, Unit) {
+
+    // Default to `Second` until proven otherwise.
```

Comment:
Why did you rewrite the logic of choosing the unit? It was much more compact before (and not using mutable variables).

Reply:
It felt weird to hard-code the `Unit::` so much :) I tried to make it more dynamic and extensible, but I you're right that we're probably concerned about trying to support a bunch more unit types.

I've switched back to the original `if/else` logic.